packages/flutter/lib/src/painting/gradient.dart

```diff
@@ -651,3 +655,232 @@ class RadialGradient extends Gradient {
+      return SweepGradient.lerp(a, this, t);
+    return super.lerpFrom(a, t);
+  }
+
+  @override
+  Gradient lerpTo(Gradient b, double t) {
+    if (b == null || (b is SweepGradient && b.colors.length == colors.length)) // TODO(ianh): remove limitation
+      return SweepGradient.lerp(this, b, t);
+    return super.lerpTo(b, t);
+  }
+
+  /// Linearly interpolate between two [SweepGradient]s.
+  ///
+  /// If either gradient is null, this function linearly interpolates from a
+  /// a gradient that matches the other gradient in [center], [startAngle],
+  /// [endAngle], [stops] and [tileMode] and with the same [colors] but
+  /// transparent (using [scale]).
+  ///
+  /// If neither gradient is null, they must have the same number of [colors].
+  ///
+  /// The `t` argument represents position on the timeline, with 0.0 meaning
+  /// that the interpolation has not started, returning `a` (or something
+  /// equivalent to `a`), 1.0 meaning that the interpolation has finished,
+  /// returning `b` (or something equivalent to `b`), and values in between
+  /// meaning that the interpolation is at the relevant point on the timeline
+  /// between `a` and `b`. The interpolation can be extrapolated beyond 0.0 and
+  /// 1.0, so negative values and values greater than 1.0 are valid (and can
+  /// easily be generated by curves such as [Curves.elasticInOut]).
+  ///
+  /// Values for `t` are usually obtained from an [Animation<double>], such as
+  /// an [AnimationController].
+  static SweepGradient lerp(SweepGradient a, SweepGradient b, double t) {
+    assert(t != null);
+    if (a == null && b == null)
+      return null;
+    if (a == null)
+      return b.scale(t);
+    if (b == null)
+      return a.scale(1.0 - t);
+    final _ColorsAndStops interpolated = _interpolateColorsAndStops(a.colors, a.stops, b.colors, b.stops, t);
+    return new SweepGradient(
+      center: AlignmentGeometry.lerp(a.center, b.center, t),
+      startAngle: math.max(0.0, ui.lerpDouble(a.startAngle, b.startAngle, t)),
+      endAngle: math.max(0.0, ui.lerpDouble(a.endAngle, b.endAngle, t)),
+      colors: interpolated.colors,
+      stops: interpolated.stops,
+      tileMode: t < 0.5 ? a.tileMode : b.tileMode, // TODO(ianh): interpolate tile mode
+    );
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (runtimeType != other.runtimeType)
+      return false;
+    final SweepGradient typedOther = other;
+    if (center != typedOther.center ||
+        startAngle != typedOther.startAngle ||
+        endAngle != typedOther.endAngle ||
+        tileMode != typedOther.tileMode ||
+        colors?.length != typedOther.colors?.length ||
+        stops?.length != typedOther.stops?.length)
+      return false;
+    if (colors != null) {
+      assert(typedOther.colors != null);
+      assert(colors.length == typedOther.colors.length);
```

Comment:
Shouldn't these both just return false instead of asserting?  If their colors are null or not the same length, then they're just not equal.

Reply:
I'm doing that here to stay consistent with the LinearGradient and RadialGradient implementations, I'm not sure why they assert on color and stops differences though. [That was introduced in this commit if it helps make sense of it: https://github.com/Hixie/flutter/commit/552f26d71324e559e69a38e5ecc16c3110c580ab#diff-1756a3cb685f68e4d8f389736fda0df2R379].

My best guess is it was added so that attempting to animated between gradients with different colors/stops would assert [although  `lerp` would assert in that case, so it seems redundant?]

Reply:
For consistency, I guess just leave them there, then, and I'll put together another PR that will remove them all after talking it over with @Hixie as to why they're in there.